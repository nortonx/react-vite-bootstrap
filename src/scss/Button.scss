.btn {
  &.btn-undefined {
    background-color: #404;
  }
}