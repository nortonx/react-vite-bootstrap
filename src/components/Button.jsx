import PropTypes from "prop-types";
import "../scss/Button.scss";

const Button = ({label, onClick, buttonType}) => {
  return(
    <>
      <button
        className={`btn btn-${buttonType ? buttonType : 'primary'}`}
        onClick={onClick}
        data-testid="button"
      >{label ? label : 'ButtonText'}</button>
    </>
  );
};

Button.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func,
  buttonType: PropTypes.string
};

export default Button;