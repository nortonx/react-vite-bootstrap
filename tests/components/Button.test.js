import { render, screen } from "@testing-library/react";
// import { BrowserRouter } from "react-router-dom";

import Button from "../../src/components/Button";

describe("Button", () => {
  it("should render and match snapshot", async () => {
    render(<Button />)
    const button = await screen.findByTestId("button");
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });
});