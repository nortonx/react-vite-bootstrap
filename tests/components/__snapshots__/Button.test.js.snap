// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and match snapshot 1`] = `
<button
  class="btn btn-primary"
  data-testid="button"
>
  ButtonText
</button>
`;
